src/components/SignUp/SignUp.tsx

```diff
@@ -0,0 +1,142 @@
+import React from 'react';
+import { Container, Paper, TextField, Button, Typography, Box, Stack } from '@mui/material';
+import { useForm } from 'react-hook-form';
+import { zodResolver } from '@hookform/resolvers/zod';
+import * as z from 'zod';
+import { useNavigate, Link } from 'react-router-dom';
+import { v4 as uuidv4 } from 'uuid';
+import type { User } from '../../types/User/types';
+
+const schema = z
+  .object({
+    id: z.string().min(1).uuid().optional(),
```

Comment:
Here as id not being modified or entered by user, there is no need to add extra validations for it because we are manually setting them by ourselves.(discussed by rinkesh bhai is yesterday's meeting)
```suggestion
    id: z.string(),
```

Reply:
done . this is my commit id a373ee732ba4f1ea52d3f51fb8c2d86512aedeb9